src/components/FeeLabelToggle/FeeLabelToggle.js

```diff
@@ -94,8 +94,8 @@ const FeeLabelToggle = ({
   const feeValue = getFormattedTransactionFeeValue(chain, txFeeInWei, gasToken);
   const currencySymbol = getCurrencySymbol(fiatCurrency);
 
-  const gasSymbol = getGasSymbol(chain, gasToken);
-  const feeInFiat = parseFloat(feeValue) * getRate(chainRates, gasSymbol, fiatCurrency);
+  const gasAddress = gasToken?.address || nativeAssetPerChain[chain].address;
+  const feeInFiat = parseFloat(feeValue) * getAssetRateInFiat(chainRates, gasAddress, fiatCurrency);
```

Comment:
Similar suggestion here, pls create a helper `function getFiatValue(balance, rate)` that will handle edge cases like NaN and nil rates.

Reply:
Not sure I follow, is this about `getAssetRateInFiat` or `feeInFiat` result?

Reply:
Ignore the original comment. What I meant is to support cases where getAssetRateInFiat return null. Probably the best way would be to have `getAssetValueInFiat(value, address, rates, currency)` to do the multiplication and edge case handling. wdyt?

Reply:
`getAssetValueInFiat` is not null and has fallback of `0`, but I guess per our last discussion let's keep as is.